content/api/commands/selectfile.md

```diff
@@ -0,0 +1,276 @@
+---
+title: selectFile
+---
+
+Selects a file or files in an HTML5 input element or simulates dragging a file
+or files into the browser.
+
+## Syntax
+
+```javascript
+.selectFile(file)
+.selectFile(file, options)
+.selectFile([file1, file2, ...])
+.selectFile([file1, file2, ...], options)
+```
+
+### Usage
+
+**<Icon name="check-circle" color="green"></Icon> Correct Usage**
+
+```javascript
+cy.get('input[type=file]').selectFile('file.json')
+cy.get('input[type=file]').selectFile(['file.json', 'file2.json'])
+cy.get('input[type=file]').selectFile({
+  contents: Buffer.from('foo'),
+  fileName: 'file.json',
+  lastModified: Date.now(),
+})
+```
+
+**<Icon name="exclamation-triangle" color="red"></Icon> Incorrect Usage**
+
+```javascript
+// Errors, cannot be chained off 'cy'
+cy.selectFile('file.json')
+
+// Will attempt to find a file called 'file contents'
+// on disk, probably not what you intended
+cy.get('input[type=file]').selectFile('file contents')
+```
+
+### Arguments
+
+**<Icon name="angle-right"></Icon> file** **_(String, Array, Object or
+Buffer)_**
+
+Either a single file, or an array of files. A file can be:
+
+- A path to a file within the project root (the directory that contains the
+  default Cypress configuration file). Eg: `'path/to/file.json'`
+- `@alias` - An alias of any type, previously stored using `.as()`. Eg:
+  `'@alias'`
+- A `Buffer()` containing binary data, such as that returned by
```

Comment:
Same question here as in the migration guide WRT `Buffer`. Are people going to understand what a `Buffer` is? Do we need to link to something here or explain this below?

Reply:
Added the same link to the node docs here.

Reply:
Does cypress provide `Buffer` for people to use, or is it something they will need to import in their spec file? It may be useful to show that in an example on this page, because it doesn't seem obvious (like, I personally do not know what I would need to do to use `Buffer` like in these examples, and would have to google it)

Reply:
Buffer is a global glass, same as Array or Object. I don't think there's anything more that needs to be done to make this explicit - the examples are complete. There's no import statement needed. If they copy paste the examples exactly as written, they will just work.

We don't explicitly call out that `cy` is provided as a global either - we just use it in our examples and assume people will understand and try it. :shrug:

Reply:
Ok, cool. As long as it works!

Reply:
I just ran a few tests with this branch. When using component testing, projects that use Webpack will polyfill `Buffer`, so your examples work as-is. However, projects that use Vite do not polyfill `Buffer`, so we end up seeing a `Buffer is not defined` error, eg.

![image](https://user-images.githubusercontent.com/54051/148593663-58f5db22-90df-4206-9a8d-d83064697b65.png)

We can work around this issue by polyfilling `Buffer` with the [buffer](https://www.npmjs.com/package/buffer) npm module (per this issue https://github.com/vitejs/vite/discussions/2785) and then doing something like this at the top of any spec files that use `Buffer`:

```js
import { Buffer } from 'buffer'
```

or globally in the `supportFile`:

```js
import { Buffer } from 'buffer'
globalThis.Buffer = Buffer
```

We should add a note to this effect somewhere!

Reply:
@BlueWinds I'm happy to take a stab at writing this, just LMK and I'll make a commit.

Reply:
Huh, today I learned that we ship with a Vite based dev server! https://docs.cypress.io/guides/component-testing/framework-configuration#Vite-Based-Projects-Vue-React

I'm getting set up with a vite project now - ideally we'd solve this in our code rather than our documentation, so that users have a consistent experience. Now I'm wondering if the readFile option to use Buffers works there at all, or if it's been broken since release and no one has noticed.

Reply:
The link you provided suggests we should be able to do `globalThis.Buffer = Buffer` just once, and it should work everywhere (including in user's spec file).

Reply:
If you need examples, also see:

- https://github.com/cypress-io/cypress-component-testing-examples/tree/main/vite-react
- https://github.com/cypress-io/cypress-component-testing-examples/tree/main/vite-vue